smartsim/_core/entrypoints/file_operations.py

```diff
@@ -0,0 +1,279 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024 Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import base64
+import collections
+import os
+import pickle
+import re
+import shutil
+from ast import literal_eval
+from distutils import dir_util  # pylint: disable=deprecated-module
+
+
+def move(parsed_args: argparse.Namespace) -> None:
+    """Move a source file or directory to another location. If dest is an
+    existing directory or a symlink to a directory, then the srouce will
+    be moved inside that directory. The destination path in that directory
+    must not already exist. If dest is an existing file, it will be overwritten.
+
+    Sample usage:
+        python _core/entrypoints/file_operations.py move
+        /absolute/file/source/path /absolute/file/dest/path
```

Comment:
One last thing! It also might be worth updating these docstrings to show that this should be used/run as a module:
```suggestion
        .. highlight:: bash
        .. code-block:: bash

            python -m smartsim._core.entrypoints.file_operations \
                move /absolute/file/source/path /absolute/file/dest/path
```